sdk/nodejs/output.ts

```diff
@@ -293,7 +333,7 @@ async function applyHelperAsync<T, U>(value: T, isKnown: boolean, isSecret: bool
     // preserve secretness from our original Output to the new one we're creating.
     return {
         value: transformed,
-        isKnown: true,
+        isKnown: isKnown || runWithUnknowns,
```

Comment:
there are three places in this code where we do this `||`.  This place, 10 lines above, and when we do `!applyDuringPreview && !runWithUnknowns` (which is equivalent to `!(isKnown || runWithUnknowns)`).  I'd prefer if we just assigned this value into a new, well-named local, and checked that.

Or, alternatively, it could even just merge them into a final 'isKnown' value that overwrites the original.  

Or, alternatively, it could be computed, then passed to a new helper funciton.